packages/evm/src/precompiles/11-bls12-map-fp2-to-g2.ts

```diff
@@ -47,6 +47,18 @@ export async function precompile11(opts: PrecompileInput): Promise<ExecResult> {
     if (opts._debug !== undefined) {
       opts._debug(`${pName} failed: ${e.message}`)
     }
+    // noble‑curves throws this for inputs that map to the point at infinity
+    if (e.message === 'bad point: ZERO') {
+      // return two zeroed field elements (x & y), each the same length as the input
+      const zeroPoint = new Uint8Array(opts.data.length * 2)
+      if (opts._debug !== undefined) {
+        opts._debug(`${pName} mapping to ZERO point, returning zero-filled output`)
+      }
+      return {
+        executionGasUsed: gasUsed,
+        returnValue: zeroPoint,
```

Comment:
Can it be explicitly checked that this should return a value and not "throw" at the EVM level? If it would throw at the EVM level the return buffer is cleared (RETURNDATASIZE would push 0 to stack) and the CALL to this precompile would push 0 to stack (instead of 1, signalling no-error for the call)

Reply:
Will come back to this. This is only throwing from within noble-curves (not for the wasm library), and aside from hardcoding results and pre-returning (therefore bypassing noble instead of waiting for it to throw) I don't see what we could do differently.